core/src/main/java/com/linecorp/armeria/server/HttpServerPipelineConfigurator.java

```diff
@@ -505,32 +514,47 @@ private void addHttpHandlers(ChannelHandlerContext ctx) {
 
         @Override
         protected void handshakeFailure(ChannelHandlerContext ctx, Throwable cause) throws Exception {
-            if (!Exceptions.isExpected(cause)) {
-                logger.warn("{} TLS handshake failed:", ctx.channel(), cause);
-            }
+            logIfUnexpected(ctx, cause);
```

Comment:
would it make sense to add a counter for handshake failures?

Reply:
That's a good idea. I'll send a follow-up PR. :bow: